strkey/main.go

```diff
@@ -118,26 +125,32 @@ func Encode(version VersionByte, src []byte) (string, error) {
 		return "", err
 	}
 
-	var raw bytes.Buffer
-
-	// write version byte
-	if err := binary.Write(&raw, binary.LittleEndian, version); err != nil {
-		return "", err
-	}
-
-	// write payload
-	if _, err := raw.Write(src); err != nil {
-		return "", err
-	}
-
-	// calculate and write checksum
-	checksum := crc16.Checksum(raw.Bytes())
-	if _, err := raw.Write(checksum); err != nil {
-		return "", err
+	// check src does not exceed maximum payload size
+	if len(src) > maxPayloadSize {
+		return "", fmt.Errorf("data exceeds maximum payload size for strkey")
 	}
 
-	result := base32.StdEncoding.WithPadding(base32.NoPadding).EncodeToString(raw.Bytes())
-	return result, nil
+	// pack
+	//  1 byte version
+	//  src bytes
+	//  2 byte crc16
+	const maxSize = 1 + maxPayloadSize + 2
+	rawArr := [maxSize]byte{}
+	size := 1 + len(src) + 2
+	raw := rawArr[:size]
+	raw[0] = byte(version)
+	copy(raw[1:], src)
+	crc := crc16.Checksum(raw[:1+len(src)])
```

Comment:
Looks like it's the same but more readable:
```suggestion
	crc := crc16.Checksum(src)
```

Reply:
It needs to calculate the checksum including the leading version byte. That's why it uses the raw string. I did try calculating the checksum in parts, first the version byte, then the src string, and I don't think the readability was better nor was there any performance benefit.

Reply:
I see, ok!